### PR TITLE
kyverno-1.10.0: new package

### LINKED ISSUE #2199: [Wolfi Package Request]: kyverno
### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

Kyverno (Greek for “govern”) is a policy engine designed specifically for Kubernetes. Some of its many features include:

- policies as Kubernetes resources (no new language to learn!)
- validate, mutate, generate, or cleanup (remove) any resource
- verify container images for software supply chain security
- inspect image metadata

Kyverno allows cluster administrators to manage environment specific configurations independently of workload configurations and enforce configuration best practices for their clusters. Kyverno can be used to scan existing workloads for best practices, or can be used to enforce best practices by blocking or mutating API requests.

Ref: https://kyverno.io/docs/introduction/

Source: https://github.com/kyverno/kyverno

Link to project: https://kyverno.io/

Package: https://github.com/kyverno/kyverno/pkgs/container/kyverno

### which versions of the software should we include?

kyverno v1.9.0 (latest)

### Add some justification for why this specific package and versions are important.

Kyverno is a very important project developing in cloud native space especially acting as a Policy Engine for Kubernetes and thus helping validate, mutate, generate, and clean up Kubernetes resources plus ensure OCI image supply chain security. It's ghcr package has more than 1.8B downloads already and is one of the fastest-growing projects with amazing involvement from the CNCF community. Currently, we already support Kyverno CLI which is "designed to validate and test policy behaviour to resources prior to adding them to a cluster." However, as the docs also say it is different from the Kyverno container image which runs as a Pod in a target Kubernetes cluster thus we can try to add the entire Kyverno package with us to make it more native experience. 

### Please check all that apply

- [X] This package has an un-restrictive license
- [X] The package/versions proposed are actively maintained upstream
- [X] I am interested in adding this package to Wolfi OS myself
- [X] I am willing to help maintain this package

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Fixes #2199 

Related: I think we can include the existing [kyverno-cli](https://github.com/wolfi-dev/os/blob/main/kyverno-cli.yaml) in this too and maybe remove the existing standalone one for more coherency.  Thoughts @imjasonh?

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [x] REQUIRED - The package is added to `packages.txt`

#### For security-related PRs

- [ ] The security fix is recorded in `advisories` and `secfixes`

#### For version bump PRs

- [ ] The `epoch` field is reset to 0
- [ ] Patch source: _patch source here_

### CODE REVIEW

**Comment by imjasonh** [comment]
> Related: I think we can include the existing [kyverno-cli](https://github.com/wolfi-dev/os/blob/main/kyverno-cli.yaml) in this too and maybe remove the existing standalone one for more coherency. Thoughts @imjasonh?

That sounds great, you can add it as a subpackage with the same name, and remove `kyverno-cli.yaml`.

File: kyverno.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - go
+
+pipeline:
+  - uses: fetch
+    with:
+      uri: https://github.com/kyverno/kyverno/archive/v${{package.version}}/v${{package.version}}.tar.gz
+      expected-sha256: dbc29ccb9a54479addc0f1c6c60879a50b9c20e0bd917cd6e0d123636f68e770
```
Comment (imjasonh) [inline kyverno.yaml:23]: I might recommend doing a git-checkout here instead. It's usually not a big difference, but some builds work better when they have the .git directory available to them, to populate things like build versions.

File: kyverno.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    with:
+      packages: ./cmd/cleanup-controller
+      ldflags: -X github.com/kyverno/kyverno/pkg/version.BuildVersion=${{package.version}}
+      output: cleanup-controller
+
+  - uses: go/build
+    with:
+      packages: ./cmd/kyverno/
+      ldflags: -X github.com/kyverno/kyverno/pkg/version.BuildVersion=${{package.version}}
+      output: kyverno
```
Comment (imjasonh) [inline kyverno.yaml:41]: These could be subpackages, named for example `kyverno-init-container`, `kyverno-cleanup-controller`, and `kyverno-cli` as you're adding. The main package can remain `kyverno` which is built from `./cmd/kyverno`.

File: kyverno.yaml (hunk 1 above)
Comment (mritunjaysharma394) [reply kyverno.yaml:23]: For sure, going to modify it

**Review by imjasonh — APPROVED** [review]

**Pull request merged**